### PR TITLE
CON-106: Updated web_http example

### PR DESCRIPTION
**Pull request by samuelraeburn**
I will do the web_socket example too when I get a chance

### CODE REVIEW

**Comment by samuelraeburn** [comment]
@alexcamposruiz I am looking into the web_socket example but have some queries for you.
One being that google no longer expose their earth API (https://stackoverflow.com/questions/42007768/google-maps-api-uncaught-error-module-earth-not-found) - the other is that maybe we need to make the .on() method public (and therefore extend it to Input)

**Comment by samuelraeburn** [comment]
@alexcamposruiz Let me know what you think of these examples. If you agree, I think we should modify the web_socket example to only show the maps (and get rid of the chart + simple versions).
With my latest commit you can publish circles, squares or triangles from Shapes demo and they will be displayed on the map (with correspond size, color, shape, location).

File: examples/nodejs/web_socket/reader_websocket.js [hunk 1]
```diff
@@ -9,60 +9,70 @@
...
+  const inputSquare = connector.getInput('MySubscriber::MySquareReader')
+  const inputTriangle = connector.getInput('MySubscriber::MyTriangleReader')
+  const inputCircle = connector.getInput('MySubscriber::MyCircleReader')
+  const io = socketsio.listen(server)
+
+  for (;;) {
+    const promisesToWait = []
+
+    // Only wait for data on an input if it has a matched publication
+    if (inputSquare.matchedPublications.length > 0) {
```
[the remark this answers is not in the document]
Comment (samuelraeburn) [reply examples/nodejs/web_socket/reader_websocket.js]: We want to make the creation of a webserver as easy as possible. Use-case is waiting for triangles + circles but not squares.

add test_no_data test 

Modify reader_websocket.io to wait on the connector and then take on each input to confirm what data has been received

Look into what bluebird library does

Worst case scenario expose connector.on

File: examples/nodejs/web_socket/README.md [hunk 2]
```diff
@@ -21,3 +20,6 @@ They can be installed using `npm install socket.io ol`
 publisher.
 * Run the web server `node ./reader_web_socket.js`
 * In a browser, navigate to [http:/localhost:7400](http://127.0.0.1:7400)
+* Select how you would like to visualise the data (simple, chart or maps).
```
Comment (alexcamposruiz) [inline examples/nodejs/web_socket/README.md]: we don't provide "chart" anymore.

File: examples/nodejs/web_socket/README.md [hunk 3]
```diff
@@ -21,3 +20,6 @@ They can be installed using `npm install socket.io ol`
...
+* Navigating to [http://localhost:7400/simple](http://localhost:7400/simple) will show the simple visualisation and
```
Comment (alexcamposruiz) [inline examples/nodejs/web_socket/README.md]: to be consistent with our docs in general, use the Am. English spelling, visualization

File: examples/nodejs/web_socket/indexSimple.html [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    x: <p style="display:inline" id="x"></p>
+    y: <p style="display:inline" id="y"></p>
+    shapesize: <p style="display:inline" id="shapesize"></p>
+    color: <p style="display:inline" id="color"></p></pre>
+  </h3>
+
+  <script src="/socket.io/socket.io.js"></script>
+
+  <script>
+    var updateGenericShapeData = function (data) {
```
Comment (alexcamposruiz) [inline examples/nodejs/web_socket/indexSimple.html]: I could be called just "updateShape"

**Pull request merged**